url64700e9ab7_dwicke_course_scheduler_tgz-pJ8-org_json_TestJ8/src/coursescheduleapp/model/CourseDatabase.java

```diff
@@ -0,0 +1,211 @@
+
+package coursescheduleapp.model;
+
+import java.rmi.RemoteException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *
+ * @author cboyle1
+ */
+public class CourseDatabase implements CourseCollectionDB
+{
+    private Connection conn;
+    /*private*/public Statement stat;
+    private String semester;
+    private BasicPublisher pub;
+    public CourseDatabase() throws Exception
+    {
+    	pub = new BasicPublisher();
+        //I have no idea what this does, but its important somehow..
+        Class.forName("org.sqlite.JDBC");
+
+        //Create connection
+        //conn = DriverManager.getConnection("jdbc:sqlite:courses.sqlite");
+        conn = DriverManager.getConnection("jdbc:sqlite:classes.sqlite");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource that might leak to a class field, causing a resource leak that slows down or crashes your system. For example, when Connection.createStatement throws an exception, the *CourseDatabase* object isn’t constructed so you can’t close this resource afterward.

**Fix**
Ensure that if an exception is thrown, the *conn* resource is closed before the constructor is exited.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).